examples/complete/main.tf

```diff
@@ -1,55 +1,343 @@
+  subnet_group_tags = {
+    Additional = "CustomSubnetGroup"
+  }
+
+  # Snapshot schedule
+  create_snapshot_schedule        = true
+  snapshot_schedule_identifier    = local.name
+  use_snapshot_identifier_prefix  = true
+  snapshot_schedule_description   = "Example snapshot schedule"
+  snapshot_schedule_definitions   = ["rate(12 hours)"]
+  snapshot_schedule_force_destroy = true
+
+  # Scheduled actions
+  create_scheduled_action_iam_role = true
+  scheduled_actions = {
+    pause = {
+      name          = "${local.name}-pause"
+      description   = "Pause cluster every night"
+      schedule      = "cron(0 22 * * ? *)"
+      pause_cluster = true
+    }
+    resize = {
+      name        = "${local.name}-resize"
+      description = "Resize cluster (demo only)"
+      schedule    = "cron(00 13 * * ? *)"
+      resize_cluster = {
+        node_type       = "ds2.xlarge"
+        number_of_nodes = 5
+      }
+    }
+    resume = {
+      name           = "${local.name}-resume"
+      description    = "Resume cluster every morning"
+      schedule       = "cron(0 12 * * ? *)"
+      resume_cluster = true
+    }
+  }
+
+  # Endpoint access
+  create_endpoint_access          = true
+  endpoint_name                   = "${local.name}-example"
+  endpoint_subnet_group_name      = aws_redshift_subnet_group.endpoint.id
+  endpoint_vpc_security_group_ids = [module.security_group.security_group_id]
+
+  # Usage limits
+  usage_limits = {
+    currency_scaling = {
+      feature_type  = "concurrency-scaling"
+      limit_type    = "time"
+      amount        = 60
+      breach_action = "emit-metric"
+    }
+    spectrum = {
+      feature_type  = "spectrum"
+      limit_type    = "data-scanned"
+      amount        = 2
+      breach_action = "disable"
+      tags = {
+        Additional = "CustomUsageLimits"
+      }
+    }
+  }
+
+  # Authentication profile
+  authentication_profiles = {
+    example = {
+      name = "example"
+      content = {
+        AllowDBUserOverride = "1"
+        Client_ID           = "ExampleClientID"
+        App_ID              = "example"
+      }
+    }
+    bar = {
+      content = {
+        AllowDBUserOverride = "1"
+        Client_ID           = "ExampleClientID"
+        App_ID              = "bar"
+      }
+    }
+  }
+
+  tags = local.tags
+}
+
+resource "aws_redshift_snapshot_copy_grant" "useast1" {
+  # Grants are declared outside of module because they are generally performed
+  # in the destination region and we do not embed multiple providers in the root module
+  provider = aws.us_east_1
+
+  snapshot_copy_grant_name = "${local.name}-us-east-1"
+  kms_key_id               = aws_kms_key.redshift_us_east_1.arn
+
+  tags = local.tags
+}
+
+################################################################################
+# Default
+################################################################################
+
+module "default" {
+  source = "../../"
+
+  cluster_identifier = "${local.name}-default"
+
```

Comment:
Let's make it clear for a user with specified `node_type`

Reply:
done